quick-dsp/src/filter/iter.rs

```diff
@@ -80,7 +80,64 @@ impl<T: Iterator, F: OneToOne<T::Item>> Iterator for OneToOneIter<T, F> {
     }
 }
 
+pub enum CrcAppendIter<T> {
```

Comment:
This is cool, but feels just a tad too magical. Would it make more sense to just have the CRC be a standalone thing?

Reply:
I think having something like this might make sense in the future and can make the code which uses the CRC easier to read. Seems nice to have.

Reply:
Okay. Let's go with this; I'll adjust.